### ISSUE TITLE
How to calculate floating-point operations(FLOPS) in mxnet?

### ISSUE THREAD

**Issue by PhanTask** [apache/incubator-mxnet#5034]
Are there any tools to calculate floating-point operations(FLOPS) in mxnet? If not, then how to do this manually? Any suggestions would be appreciated.

**Comment by szha** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to ping me to reopen if this is still an active issue. Thanks!

**Comment by ChaiBapchya** [comment]
@szha can we reopen this issue?
In one of my big data assignments, I was tasked to find "number of floating point computations taking place when one single data is passed through the network"
Is there a way to find this directly in mxnet?

**Comment by szha** [comment]
@ChaiBapchya there's no existing feature in mxnet for calculating flops and people usually estimate this based on manual calculation on different (mostly gemm type) operators.

**Comment by lupesko** [comment]
@mxnet-label-bot add [Feature request]

**Comment by ChaiBapchya** [comment]
I was able to count total number of parameters for a model in Pytorch using this command

```
pytorch_total_params = sum(p.numel() for p in model.parameters() if p.requires_grad)
```
Source - https://discuss.pytorch.org/t/how-do-i-check-the-number-of-parameters-of-a-model/4325/9?u=chaibapchya

**Comment by Ldpe2G** [comment]
@ChaiBapchya I wrote a tool to calculate flops and model size of a given mxnet symbol, you can have a try.
https://github.com/Ldpe2G/DeepLearningForFun/tree/master/MXNet-Python/CalculateFlopsTool
